src/utilities/dataplane.spec.ts

```diff
@@ -0,0 +1,171 @@
+      {
+        message: 'undefined subscriptions is offline',
+        args: [{}],
+        expected: 'offline',
+      },
+      {
+        message: 'single connected subscription is online',
+        args: [{
+          subscriptions: [
+            {
+              connectTime: '1',
+            } as DiscoverySubscription,
+          ],
+        }],
+        expected: 'online',
+      },
+      {
+        message: 'single disconnected subscription is offline',
+        args: [{
+          subscriptions: [
+            {
+              connectTime: '1',
+              disconnectTime: '2',
+            } as DiscoverySubscription,
+          ],
+        }],
+        expected: 'offline',
+      },
+      {
+        message: 'multiple subscriptions with a single connected is online',
+        args: [{
+          subscriptions: [
+            {
+              connectTime: '1',
+            } as DiscoverySubscription,
+            {
+              connectTime: '1',
+              disconnectTime: '2',
+            } as DiscoverySubscription,
+          ],
+        }],
+        expected: 'online',
+      },
+    ] as TestCases))('$message', (item) => {
+      expect(getItemStatusFromInsight(...item.args)).toBe(item.expected)
+    })
+  })
+
+  describe('getStatusAndReason', () => {
+    type TestCases = {message: string, args: [{ networking: DataPlaneNetworking }, { subscriptions?: DiscoverySubscription[] }], expected: any}[]
+    test.each(([
+      {
+        message: 'nothing defined is online',
+        args: [{
+          networking: {
+            address: '',
+          },
+        }],
+        expected: {
+          status: 'online',
+          reason: [],
+        },
+      },
+      {
+        message: 'empty inbounds is online',
+        args: [{
+          networking: {
+            address: '',
+            inbound: [],
+          },
+        }],
+        expected: {
+          status: 'online',
+          reason: [],
+        },
+      },
+      {
+        message: 'single healthy inbound defers to getItemStatusFromInsight (online)',
+        args: [{
+          networking: {
+            address: '',
+            inbound: [{
+              health: {
+                ready: true,
+              },
+            }],
+          },
+        }, {
+          subscriptions: [
+            {
+              connectTime: '1',
+            },
+          ],
+        }],
+        expected: {
+          status: 'online',
+          reason: [],
+        },
+      },
+      {
+        message: 'single unhealthy inbound is offline with a single reason',
+        args: [{
+          networking: {
+            address: '',
+            inbound: [{
+              health: {
+                ready: false,
+              },
+              port: 1,
+              tags: {
+                'kuma.io/service': 'service',
+              },
+            }],
+          },
+        }],
+        expected: {
+          status: 'offline',
+          reason: [''],
```

Comment:
nit: Would it be more readable if reason was `reasonCount` or something like this? It's kind of weird to see this here and then see it works because we assert on length of the array

Reply:
Yeah maybe, I started off trying to keep it the same as shape of the return from the function, and then realized these were strings that needed more mock data adding to get the correct string, then I kinda figured "wait a second I don't want to test the string content of the `reason`s, just that there are the right amount", and left it like this.

Me personally I'd keep `expected` to be the same shape as the return from the function, but I don't mind changing it to `reasonLength` if you would prefer? Let me know what you think after reading this, happy to change if you are still keen.

Reply:
I don't mind either way really. Just pointing out it's reading these tests put me off-guard a little

Reply:
```suggestion
          reason: ['fake error message to fill the length of `reason`'],
```
Would something like that ^ have helped? Means we keep the shape and it might be clearer when reading it?

Reply:
Though it feels hacky

Reply:
Oh I just realized I've left the the correct values in from when I decided not to do it that way 😆 I'm half way there, one sec

Reply:
LG